Core/include/Acts/Utilities/AngleHelpers.hpp

```diff
@@ -9,26 +9,69 @@
 #pragma once
 
 #include <cmath>
+#include <concepts>
+#include <numbers>
 
 namespace Acts::AngleHelpers {
 
+template <std::floating_point Scalar>
+struct EtaThetaConversionTraits {
+  static constexpr Scalar minTheta = 0;
+  static constexpr Scalar maxTheta = std::numbers::pi_v<Scalar>;
+
+  static constexpr Scalar minEta = -std::numeric_limits<Scalar>::infinity();
+  static constexpr Scalar maxEta = std::numeric_limits<Scalar>::infinity();
+};
```

Comment:
I wonder if this may be a hostage to fortune if we have other `std::floating_point` types in future (eg. `long double`, `std::float16_t`, GPU types). Those conversions could still return `nan` rather than `inf`. I think any use of this template that is not specialized could be problematic.

Reply:
https://github.com/acts-project/acts/blob/6797a56a15e67cc036c15cfc623558d2e0286990/Core/include/Acts/Utilities/AlgebraHelpers.hpp#L217-L250

I guess we can do it like here and force a compile error in this case